### ISSUE TITLE
SDL_SetWindowSize and element.requestFullscreen() troubles

### ISSUE THREAD

**Issue by parasti** [emscripten-ports/SDL2#138]
I figured I could this:

```c
    double clientWidth, clientHeight;

    emscripten_get_element_css_size("#canvas", &clientWidth, &clientHeight);

    int w = (int) clientWidth;
    int h = (int) clientHeight;

    if (w != window_w || h != window_h)
        SDL_SetWindowSize(window, w, h);
```

to keep my canvas resolution in sync with the CSS dimensions. Doing it via SDL_SetWindowSize makes sure SDL2 doesn't miscalculate things like mouse coordinates.

So far, so good.

The problem is, I also want to handle fullscreen via element.requestFullscreen() instead of going through SDL_SetWindowFullscreen and emscripten_request_fullscreen_strategy.

Why? Because emscripten_request_fullscreen_strategy does many things, none of which are the thing that I want. I just want a fullscreen canvas, without the inline styles, layers of compatiblity logic and many possible points of failure.

Unfortunately, element.requestFullscreen() still ends up going through there.

Doing element.requestFullscreen() on any element in the document triggers a fullscreenchange event, for which SDL2 installs a callback. This callback sets the SDL_WINDOW_FULLSCREEN flag on the window, which causes SDL_SetWindowSize to change behavior: it instead does a SDL_UpdateFullscreenMode, again, ending up at Emscripten_SetWindowFullscreen and emscripten_request_fullscreen_strategy.

Which was what I was attempting to avoid in the first place.

TLDR; wondering how can I achieve these three things together:

 * match canvas resolution and CSS size
 * while having SDL2 translate mouse coordinates correctly
 * and using element.requestFullscreen for fullscreen while bypassing emscripten_request_fullscreen_strategy entirely

I also just realized that Emscripten_HandleFullscreenChange sets a variable to zero, and then checks if that variable is zero (https://github.com/emscripten-ports/SDL2/blob/master/src/video/emscripten/SDL_emscriptenevents.c#L599). Could that be the culprit?

```c
        window_data->requested_fullscreen_mode = 0;

        if(!window_data->requested_fullscreen_mode)
            window_data->window->flags |= SDL_WINDOW_FULLSCREEN; /*we didn't request fullscreen*/
```

**Comment by parasti** [comment]
Changed that part to this and it works very well, but not sure of the repercussions (if any).

```c
    if(fullscreenChangeEvent->isFullscreen)
    {
        window_data->window->flags |= window_data->requested_fullscreen_mode;

        window_data->requested_fullscreen_mode = 0;
    }

```

**Comment by Daft-Freak** [comment]
Hmm, the only thing that changes is that the fullscreen flag won't be set on the window if you didn't request fullscreen through SDL... I suppose if you're setting fullscreen from outside SDL that's what you want anyway.
